### PR TITLE
Add Deposit Feature

### PR DESCRIPTION
**Pull request by xiaoge26**
Added deposit feature and handled number format and null pointer exceptions.

### CODE REVIEW

**Comment by xiaoge26** [comment]
I'll fix the Checkstyle errors now.

File: src/main/java/seedu/bankwithus/Parser.java [hunk 1]
```diff
@@ -19,15 +19,14 @@ public float parseWithdrawAmt(String args) throws NegativeAmountException {
         if (withdrawAmt < 0) {
             throw new NegativeAmountException();
         }
-        float currBal = bwu.accounts.accounts.get(0).balance;
+        float currBal = bwu.getAccountList().accounts.get(0).balance;
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/Parser.java:22]: Wait doesn't the getCurrentAccount method that you added do the exact same thing as accounts.get(0)? I think you can use that method instead

**Comment by xiaoge26** [comment]
I rewrited the tests for accountList class as well.

File: src/main/java/seedu/bankwithus/Parser.java [hunk 2]
```diff
@@ -38,15 +37,26 @@ public void parseUserInput(String input) throws CommandNotFoundException {
         case "exit":
             bwu.isExitEntered = true;
             break;
+        case "deposit":
+            try {
+                bwu.getAccountList().depositMoney(args);
+                screen.showDepositMessage();
+                screen.showBal(bwu.getAccountList().accounts.get(0).getAccountBalance());
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/Parser.java:44]: Same as comment 3 here, you can use getCurrentAccount for this

File: data/save.txt [hunk 3]
```diff
@@ -1 +1 @@
-jen;0.0
```
Comment (vishnuvk47) [inline data/save.txt:1]: whenever possible, I would recommend to delete the data/save.txt before pushing as well.

Comment (xiaoge26) [reply data/save.txt:1]: OK! I will do.

File: src/main/java/seedu/bankwithus/Parser.java [hunk 4]
```diff
@@ -38,15 +37,26 @@ public void parseUserInput(String input) throws CommandNotFoundException {
...
-            String accDetails = bwu.accounts.getAllAccountDetails();
+            String accDetails = bwu.getAccountList().getAllAccountDetails();
             screen.viewAccount(accDetails);
             break;
         case "withdraw":
             try {
                 float finalBal = parseWithdrawAmt(args);
                 if(finalBal >= 0) {
-                    bwu.accounts.accounts.get(0).setBalance(finalBal);
+                    bwu.getAccountList().accounts.get(0).setBalance(finalBal);
```
[the remark this answers is not in the document]
Comment (xiaoge26) [reply src/main/java/seedu/bankwithus/Parser.java]: Ok sure!

File: src/main/java/seedu/bankwithus/AccountList.java [hunk 5]
```diff
@@ -46,5 +41,12 @@ public String getAllAccountDetails() {
         return temp;
     }
 
+    public Account getCurrentAccount() {
+        return accounts.get(0);
+    }
 
+    public void depositMoney(String depositAmountString) throws NumberFormatException, NullPointerException {
+        float depositAmount = Float.parseFloat(depositAmountString);
```
[the remark this answers is not in the document]
Comment (xiaoge26) [reply src/main/java/seedu/bankwithus/AccountList.java:52]: Exception added.

File: src/main/java/seedu/bankwithus/BankWithUs.java [hunk 6]
```diff
@@ -94,6 +100,15 @@ public void run() throws IOException {
...
     }
 
+
+    public AccountList getAccountList() {
+        return accountList;
+    }
+
+    public void setAccountList(AccountList accountList) {
+        this.accountList = accountList;
+    }
```
[the remark this answers is not in the document]
Comment (xiaoge26) [reply src/main/java/seedu/bankwithus/BankWithUs.java]: OK I deleted it!

**Review by tyuyang — APPROVED** [review]

**Pull request merged**